sql/core/src/test/scala/org/apache/spark/sql/test/SharedSparkSession.scala

```diff
@@ -79,6 +80,15 @@ trait SharedSparkSessionBase
       StaticSQLConf.WAREHOUSE_PATH,
       conf.get(StaticSQLConf.WAREHOUSE_PATH) + "/" + getClass.getCanonicalName)
     conf.set(StaticSQLConf.LOAD_SESSION_EXTENSIONS_FROM_CLASSPATH, false)
+    // SPARK-51365: Due to the fact that the GitHub-hosted Runner on the `macOS + AppleSilicon chip`
+    // combination has only half the memory resources of Runners with other specifications,
+    // it is necessary to limit the number of threads in this case to avoid the issue of
+    // memory resources being exhausted by frequent thread creation during testing.
+    if (sys.env.contains("GITHUB_ACTIONS") && Utils.isMacOnAppleSilicon) {
```

Comment:
Since similar issues haven't occurred in other pipelines, in this PR I've only increased the limit on the number of threads for the currently failing test scenario.

cc @dongjoon-hyun Do you think this is ok?